File: leetcode/tree/938.range-sum-of-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def traverse(self,root,l,r,s):
        if not root:
            return s
        
        if root.val >= l and root.val <= r:
            s += root.val
        s = self.traverse(root.left,l,r,s)
        s = self.traverse(root.right,l,r,s)
            
        return s
    
    
    def rangeSumBST(self, root: TreeNode, L: int, R: int) -> int:
        if not root:
            return 0
        s = 0
        if L > R:
            L,R = R,L
            
        return self.traverse(root,L,R,s)

File: leetcodeJS/tree/938.range-sum-of-bst.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} L
 * @param {number} R
 * @return {number}
 */
var rangeSumBST = function (root, L, R) {
  if (!root) return 0;
  cnt = 0;
  if (root.val >= L && root.val <= R) {
    cnt += root.val;
  }

  cnt += rangeSumBST(root.left, L, R);
  cnt += rangeSumBST(root.right, L, R);
  return cnt;
};
